WordPress/Classes/Utility/DisplayableImageHelper.m

```diff
@@ -0,0 +1,184 @@
+    if ([attachments count] == 0) {
+        return nil;
+    }
+
+    NSString *imageToDisplay;
+
+    attachments = [self filteredAttachmentsArray:attachments];
+
+    NSDictionary *attachment = [attachments firstObject];
+    NSInteger width = [[attachment numberForKey:@"width"] integerValue];
+    if (width >= FeaturedImageMinimumWidth) {
+        imageToDisplay = [attachment stringForKey:@"URL"];
+    }
+
+    return imageToDisplay;
+}
+
++ (NSArray *)filteredAttachmentsArray:(NSArray *)attachments
+{
+    NSString *key = @"mime_type";
+    NSPredicate *predicate = [NSPredicate predicateWithFormat:@"%K BEGINSWITH %@", key, @"image"];
+    attachments = [attachments filteredArrayUsingPredicate:predicate];
+
+    return [attachments sortedArrayUsingComparator:^NSComparisonResult(id obj1, id obj2) {
+        NSDictionary *attachmentA = (NSDictionary *)obj1;
+        NSDictionary *attachmentB = (NSDictionary *)obj2;
+        NSString *key = @"width";
+        NSNumber *widthA = [attachmentA numberForKey:key] ?: @(0);
+        NSNumber *widthB = [attachmentB numberForKey:key] ?: @(0);
+
+        if ([widthA integerValue] < [widthB integerValue]) {
+            return NSOrderedDescending;
+        } else if ([widthA integerValue] > [widthB integerValue]) {
+            return NSOrderedAscending;
+        } else {
+            return NSOrderedSame;
+        }
+    }];
+}
+
+/**
+ Search the passed string for an image that is a good candidate to feature.
+
+ @param content The content string to search.
+ @return The url path for the image or an empty string.
+ */
++ (NSString *)searchPostContentForImageToDisplay:(NSString *)content
+{
+    NSString *imageSrc = @"";
+    // If there is no image tag in the content, just bail.
+    if (!content || [content rangeOfString:@"<img"].location == NSNotFound) {
+        return imageSrc;
+    }
+
+    // Get all the things
+    static NSRegularExpression *imgRegex;
+    static NSRegularExpression *srcRegex;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        NSError *error;
+        imgRegex = [NSRegularExpression regularExpressionWithPattern:@"<img(\\s+.*?)(?:src\\s*=\\s*(?:'|\")(.*?)(?:'|\"))(.*?)>" options:NSRegularExpressionCaseInsensitive error:&error];
+        srcRegex = [NSRegularExpression regularExpressionWithPattern:@"src\\s*=\\s*(?:'|\")(.*?)(?:'|\")" options:NSRegularExpressionCaseInsensitive error:&error];
+    });
+
+    NSArray *matches = [imgRegex matchesInString:content options:NSRegularExpressionCaseInsensitive range:NSMakeRange(0, [content length])];
+
+    NSInteger currentMaxWidth = FeaturedImageMinimumWidth;
+    for (NSTextCheckingResult *match in matches) {
+        NSString *tag = [content substringWithRange:match.range];
+        // Get the source
+        NSRange srcRng = [srcRegex rangeOfFirstMatchInString:tag options:NSRegularExpressionCaseInsensitive range:NSMakeRange(0, [tag length])];
```

Comment:
It wouldn't really break the code.  It'd just trim it down a bit to separate concerns, improve readability, etc.

Just give this one more look to see if it makes sense for you - if it doesn't you can keep it as it is now.

```objective-c
for (NSTextCheckingResult *match in matches) {
    NSString *tag = [content substringWithRange:match.range];
    // Get the src of the image, triming off everything but the url itself.
    NSRange srcRng = [srcRegex rangeOfFirstMatchInString:tag options:NSRegularExpressionCaseInsensitive range:NSMakeRange(0, [tag length])];
    NSString *src = [tag substringWithRange:srcRng];
    NSCharacterSet *charSet = [NSCharacterSet characterSetWithCharactersInString:@"\"'="];
    NSRange quoteRng = [src rangeOfCharacterFromSet:charSet];
    src = [src substringFromIndex:quoteRng.location];
    src = [src stringByTrimmingCharactersInSet:charSet];
    // Check the tag for a good width
    NSInteger width = MAX([self widthFromElementAttribute:tag], [self widthFromQueryString:src]);
    if (width > currentMaxWidth) {
        imageSrc = src;
        currentMaxWidth = width;
    }
}
```

...would become...

```objective-c
for (NSTextCheckingResult *match in matches) {
    NSString *tag = [content substringWithRange:match.range];
    NSString* src = [self sourceFromTag:tag];

    // Check the tag for a good width
    NSInteger width = MAX([self widthFromElementAttribute:tag], [self widthFromQueryString:src]);
    if (width > currentMaxWidth) {
        imageSrc = src;
        currentMaxWidth = width;
    }
}
```